code/modules/reagents/reagent_containers/chem_heirloom.dm

```diff
@@ -0,0 +1,46 @@
+//Chemist's heirloom
+
+/obj/item/reagent_containers/glass/chem_heirloom
+	name = "hard locked bottle of"
+	desc = "A hard locked bottle of"
+	volume = 100
+	spillable = FALSE
+	reagent_flags = NONE
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "hard_locked_closed"
+	item_state = "hard_locked_closed"
+	fill_icon_thresholds = list(0, 10, 25, 50, 75, 80, 90)
+	var/locked = TRUE
+	var/datum/callback/roundend_callback
+	
+/obj/item/reagent_containers/glass/chem_heirloom/Initialize(mapload, vol)
+	..()
+	update_icon()
+	var/datum/reagent/R = get_unrestricted_random_reagent_id()
+	name ="[name] [initial(R.name)]"
+	reagents.add_reagent(R, volume)
+	var/datum/component/heirloom/H = GetComponent(/datum/component/heirloom)
+	desc = H ? "The [H.family_name] family's long-cherished wish is to open this bottle and get its chemical outside. Can you make that wish come true?" : "[desc] [initial(R.name)]."
```

Comment:
Only humans have family names. It should have another desc for non-human species.
I think I would try this
```
var/speciesdesc = ishuman(H) ? "The [H.family_name] family's" : "Your family's"
desc = H ? "[speciesdesc] long-cherished..."
```